compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/codegen/irCodegenUtils.kt

```diff
@@ -83,11 +83,10 @@ internal val DeclarationDescriptorWithSource.psiElement: PsiElement?
 fun JvmBackendContext.getSourceMapper(declaration: IrClass): DefaultSourceMapper {
     val sourceManager = this.psiSourceManager
     val fileEntry = sourceManager.getFileEntry(declaration.fileParent)
-    check(fileEntry != null) { "No PSI file entry found for class: ${declaration.dump()}" }
```

Comment:
Could you clarify this change?

Reply:
Otherwise, FunctionCodegen cannot create per-function SMAPs if the class is in a synthetic file (a multifile facade).

I actually don't know what `getFileEntry` is for: synthetic files *do* have entries, which are accessible as `declaration.fileParent.fileEntry`, and I might be wrong, but I believe for non-synthetic files `getFileEntry(fileParent)` and `fileParent.fileEntry` should be the same.

Reply:
Looks like we need smap test with multifile facades (ideally with 2 parts each with inline method but I'm not sure that framework could support such thing).

Reply:
Not sure what the point is, since at least right now nothing is inlined into multifile facades and thus they do not have SMAPs (IrSourceCompilerForInline never sets writeSourceMap to true).

Reply:
Yes, but i just want to test that part ones would be presented. 

Looks like there was nonsence in 'ideally with 2 parts each with inline method but I'm not sure that framework could support such thing'

Reply:
@pyos Maybe I'm a little overweight here, sorry.